### PR TITLE
8106 cesium viewer settings

### PR DESCRIPTION
**Pull request by weberjavi**
## Description

This PR creates `MapSettings` component that is used in the `Map` plugin inside the `Settings` `container`. With this approach whenever a user uses the `Map` plugin the settings window will display the available settings for each map type.
In this first iteration only `cesium` map type settings are available.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Feature

## Issue

**What is the current behavior?**
No option to update cesium scene options whatsoever. Some of them might be critical for the proper display of 3D layers on top of the terrain (see #8058 as an example).

#8106 

**What is the new behavior?**

Some cesium scene configuration options have been added to the settings panel (displayed only when the selected map type is cesium). 
The added options (`showSkyAtmosphere`, `showGroundAtmosphere`, `enableFog`, `depthTestAgainstTerrain`) are all of `boolean` type. 

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

## Other useful information
In order to provide some configuration to the scene lighting a different type of form inputs need to be designed/implemented. [Scene lights](https://cesium.com/learn/cesiumjs/ref-doc/Light.html) could update its color and intensity but some sort of color picker and numeric input should be used.

### CODE REVIEW

File: web/client/plugins/map/mapsettings/MapSettings.jsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    isCesium,
+    updateConfigAction
+}) => {
+    const [mapOptions, setMapOptions] = useState({});
+
+    useEffect(() => {
+        if (map) {
+            setMapOptions({
+                ...ConfigUtils.getConfigProp("defaultMapOptions")[mapType],
+                ...map.mapOptions
```
Comment (allyoucanmap) [inline web/client/plugins/map/mapsettings/MapSettings.jsx:43]: Here is missing the Map.cfg.mapOptions default configuration If compared with options merge inside the Map plugin. We could apply this change inside the Settings plugin

```diff
        return Object.keys(settingsFirst)
            .filter(this.isEnabled)
            .map((setting) => settingsFirst[setting])
-           // TODO: here every item (item.tool) we emit should have a "key" property
-           .concat(this.props.items.map((item) => item.tool))
+           .concat(this.props.items.map((item) =>
+               castArray(item.tool)
+                   .map((tool, idx) => cloneElement(tool, { ...item.cfg, key: `${item.name}-${idx}` })))
+           )
            .concat(
                Object.keys(settingsLast)
                    .filter(this.isEnabled)
                    .map((setting) => settingsLast[setting])
            );
```

- cloneElement from react
- castArray from lodash

with item.cfg spreat to the tool props we will get mapOptions from cfg inside this component so we can merge it with the other map options. Note the mapOptions from cfg is grouped with mapType

File: web/client/actions/map.js [hunk 2]
```diff
@@ -31,6 +31,7 @@ export const MOUSE_OUT = 'MOUSE_OUT';
 export const MAP_PLUGIN_LOAD = 'MAP:MAP_PLUGIN_LOAD';
 export const ORIENTATION = 'MAP:ORIENTATION';
 export const UPDATE_MAP_VIEW = 'MAP:UPDATE_MAP_VIEW';
+export const UPDATE_SCENE_CONFIG = 'MAP:UPDATE_SCENE_CONFIG';
```
Comment (allyoucanmap) [inline web/client/actions/map.js:34]: Could we call this `UPDATE_MAP_OPTIONS` instead? Looking to the reducer it seems not strictly related to the scene so I would suggest to keep this action related to all the map options.

File: web/client/actions/map.js [hunk 3]
```diff
@@ -254,6 +255,11 @@ export const updateMapView = (data) => ({
     data
 });
 
+export const updateSceneConfig = (configUpdate) => ({
```
Comment (allyoucanmap) [inline web/client/actions/map.js:258]: following the previous comment also this should be changed to updateMapOptions

File: web/client/components/map/cesium/Map.jsx [hunk 4]
```diff
@@ -190,6 +190,19 @@ class CesiumMap extends React.Component {
...
+        }
+        if (prevProps && (!isEqual(this.props.mapOptions.showGroundAtmosphere, prevProps?.mapOptions?.showGroundAtmosphere))) {
+            this.map.scene.globe.showGroundAtmosphere = this.props.mapOptions.showGroundAtmosphere;
+        }
+        if (prevProps && (!isEqual(this.props.mapOptions.enableFog, prevProps?.mapOptions?.enableFog))) {
+            this.map.scene.fog.enabled = this.props.mapOptions.enableFog;
+        }
+        if (prevProps && (!isEqual(this.props.mapOptions.depthTestAgainstTerrain, prevProps?.mapOptions?.depthTestAgainstTerrain))) {
+            this.map.scene.globe.depthTestAgainstTerrain = this.props.mapOptions.depthTestAgainstTerrain;
+        }
```
Comment (allyoucanmap) [inline web/client/components/map/cesium/Map.jsx:205]: Please update this branch with the latest master and verify if we need to add map.scene.requestRender(); after this scene updates to visualize the changes in the viewer

related to https://github.com/geosolutions-it/MapStore2/commit/3df6f48a01f34b4dbd9ef444a0a30b6ae819187f

File: web/client/components/map/cesium/Map.jsx [hunk 5]
```diff
@@ -190,6 +190,19 @@ class CesiumMap extends React.Component {
             };
             this.setView(position);
         }
+
+        if (prevProps && (!isEqual(this.props.mapOptions.showSkyAtmosphere, prevProps?.mapOptions?.showSkyAtmosphere))) {
```
Comment (allyoucanmap) [inline web/client/components/map/cesium/Map.jsx:194]: I'm not sure we need is equal for this options comparison, shouldn't be all boolean? same for these others new props

**Review by allyoucanmap — CHANGES_REQUESTED** [review]

File: web/client/components/map/cesium/Map.jsx (hunk 4 above)
Comment (weberjavi) [reply web/client/components/map/cesium/Map.jsx:210]: It is not needed, it seems that the scene is updated (and rendered) right after property reassignment.

**Review by allyoucanmap — APPROVED** [review]

**Comment by allyoucanmap** [comment]
@ElenaGallo please test this fix on dev, thanks

**Pull request merged**